packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -203,18 +205,29 @@ def _ai_env_status_for_provider_text(self, provider_id):
         ):
             return ""  # No message necessary
 
+        # Avoid composing strings, to make localization easier in the future
+        prefix = "Requires environment variable:"
+        var_names = []
         try:
-            env_var = self.providers[provider_id].auth_strategy.name
+            var_names.append(self.providers[provider_id].auth_strategy.name)
         except AttributeError:  # No "name" attribute
-            return ""
-
-        output = f"Requires environment variable {env_var} "
-        if os.getenv(env_var) != None:
-            output += "(set)"
-        else:
-            output += "(not set)"
+            # Try multiple names
+            try:
+                var_names = self.providers[provider_id].auth_strategy.names
+                prefix = "Requires environment variables:"
+            except AttributeError:  # No "names" attribute
+                return ""
+
+        annotated_var_names = []
+        for var_name in var_names:
+            annotated_var_name = var_name
+            if var_name in os.environ:
+                annotated_var_name += " (set)"
+            else:
+                annotated_var_name += " (not set)"
+            annotated_var_names.append(annotated_var_name)
 
-        return output + "\n"
+        return prefix + " " + ", ".join(annotated_var_names) + "\n"
```

Comment:
1. Extract `prefix` to a top-level constant as advised above.
2. This method is way too verbose for what it is trying to accomplish. Consider this block:

```py
# only handle providers with "env" or "multienv" auth strategy
auth_strategy = getattr(self.providers[provider_id], 'auth_strategy', None)
if not auth_strategy or (auth_strategy.type != "env" and auth_strategy.type != "multienv"):
    return ""

envvars = [auth_strategy.name if auth_strategy.type == "env" else *auth_strategy.names]
for i in range(len(envvars)):
    envvars[i] += " (set)" if envvars[i] in os.environ else " (not set)"

return prefix + " " + ",".join(envvars) + "\n"
```